docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -0,0 +1,87 @@
+# SPIFFE Compliance
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/13659
+
+## Context and Problem Statement
+
+In today’s service mesh systems, secure communication between services is essential. To make that possible, each service needs a clear and trusted identity: so we know exactly who is talking to whom.
+
+One of the most trusted standards for defining these service identities is SPIFFE (Secure Production Identity Framework for Everyone). It gives us:
+
+* A standard way to name services: `spiffe://<trust-domain>/<path>`
+* A strict format for certificates, especially the Subject Alternative Name (SAN) - which must contain exactly one URI (the SPIFFE ID)
+* Tools and rules for issuing, rotating, and checking these identities
+
+Right now, Kuma does something similar by generating SPIFFE-like IDs like `spiffe://<mesh-name>/<service-name>`. But it also adds extra SANs (like `kuma://kuma.io/version/v1`) based on tags. These extra SANs make Kuma's certificates non-compliant with SPIFFE, meaning they can’t be used directly with SPIFFE-based systems like SPIRE, or other tools that expect full compliance.
+
+In SPIFFE, a Trust Domain defines which services are allowed to trust each other. It’s like a boundary — services inside the same trust domain can safely communicate. To talk across trust domains (for example, between two companies or clusters), you need federation, which lets them trust each other's identities.
+
+Kuma currently uses the mesh name as the trust domain. While this works inside Kuma, it doesn’t clearly separate trust boundaries, and it limits how easily Kuma can connect with external services or other applications in the same trust domain.
+
+This setup causes problems when:
+
+* You want to connect Kuma with other SPIFFE-compliant systems outside the mesh
+* You run multiple clusters that need to trust each other
```

Comment:
I'm not sure I 100% understand this - currently if you run multiple clusters (zones) and they share the same `mesh` then everything is fine right?

Reply:
Yes, but all have the same trust domain which might not be correct since each cluster can have a different one

Reply:
So what you want is to have "zones" that have different trustDomains.